src/contracts/middleware/BLSIndexRegistryCoordinator.sol

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "../interfaces/IRegistryCoordinator.sol";
+import "../interfaces/IBLSPubkeyRegistry.sol";
+import "../interfaces/IIndexRegistry.sol";
+
+import "../libraries/BytesArrayBitmaps.sol";
+
+import "./StakeRegistry.sol";
+
+contract BLSIndexRegistryCoordinator is StakeRegistry, IRegistryCoordinator {
+    using BN254 for BN254.G1Point;
+
+    /// @notice the BLS Pubkey Registry contract that will keep track of operators' BLS public keys
+    IBLSPubkeyRegistry public immutable blsPubkeyRegistry;
+    /// @notice the Index Registry contract that will keep track of operators' indexes
+    IIndexRegistry public immutable indexRegistry;
+    /// @notice the mapping from operator's operatorId to the bitmap of quorums they are registered for
+    mapping(bytes32 => uint256) public operatorIdToQuorumBitmap;
+    /// @notice the mapping from operator's address to the operator struct
+    mapping(address => Operator) public operators;
```

Comment:
probably want to:

- include these in the interface
- make the `operators` mapping _internal_ (so prepend an underscore) and write an external/public getter function that returns a struct in memory (since otherwise the automatic getter will return a tuple instead of a struct)

Reply:
made this change